src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -1,690 +1,677 @@
-/* eslint-disable */
-
-/* fork of Codemirror's show-hint addon */
-// eslint-disable
+// fork of Codemirror's show-hint addon
 // CodeMirror, copyright (c) by Marijn Haverbeke and others
 // Distributed under an MIT license: http://codemirror.net/LICENSE
 
-(function (mod) {
-  if (typeof exports === 'object' && typeof module === 'object') // CommonJS
-  { mod(require('codemirror')); } else if (typeof define === 'function' && define.amd) // AMD
-  { define(['codemirror'], mod); } else // Plain browser env
-  { mod(CodeMirror); }
-}((CodeMirror) => {
-  const HINT_ELEMENT_CLASS = 'CodeMirror-hint';
-  const ACTIVE_HINT_ELEMENT_CLASS = 'CodeMirror-hint-active';
-
-  // This is the old interface, kept around for now to stay
-  // backwards-compatible.
-  CodeMirror.showHint = function (cm, getHints, options) {
-    if (!getHints) return cm.showHint(options);
-    if (options && options.async) getHints.async = true;
-    const newOpts = { hint: getHints };
-    if (options) for (const prop in options) newOpts[prop] = options[prop];
-    return cm.showHint(newOpts);
-  };
+import CodeMirror from 'codemirror'; // eslint-disable-line import/no-extraneous-dependencies
+import {
+  get,
+  isEmpty,
+  forOwn
+} from 'lodash';
+
+const HINT_ELEMENT_CLASS = 'CodeMirror-hint';
+const ACTIVE_HINT_ELEMENT_CLASS = 'CodeMirror-hint-active';
+
+CodeMirror.registerHelper('hint', 'auto', { resolve: resolveAutoHints });
+
+const defaultOptions = {
+  hint: CodeMirror.hint.auto,
+  completeSingle: true,
+  alignWithWord: true,
+  closeCharacters: /[\s()[]{};:>,]/,
+  closeOnUnfocus: true,
+  completeOnSingleClick: true,
+  container: null,
+  customKeys: null,
+  extraKeys: null
+};
+
+// This is the old interface, kept around for now to stay backwards-compatible
+CodeMirror.showHint = function (cm, getHints, options) {
+  if (!getHints) return cm.showHint(options);
+  if (options && options.async) getHints.async = true;
+  const newOpts = Object.assign({ hint: getHints }, options);
+
+  return cm.showHint(newOpts);
+};
+
+CodeMirror.defineExtension('showHint', function (initialOptions) {
+  const options = parseOptions(this, this.getCursor('start'), initialOptions);
+  const selections = this.listSelections();
+
+  if (selections.length > 1) return;
+  // By default, don't allow completion when something is selected.
+  // A hint function can have a `supportsSelection` property to
+  // indicate that it can handle selections.
+  if (this.somethingSelected()) {
+    if (!options.hint.supportsSelection) return;
+    // Don't try with cross-line selections
+    for (let i = 0; i < selections.length; i++) if (selections[i].head.line !== selections[i].anchor.line) return;
```

Comment:
I think the next format should be more readable: 

` for (let i = 0; i < selections.length; i++) {
     if (selections[i].head.line !== selections[i].anchor.line) return;
  }
`
What do you think about it?